dataplug-spotify/app/org/hatdex/dataplugSpotify/apiInterfaces/SpotifyProfileCheck.scala

```diff
@@ -48,9 +50,34 @@ class SpotifyProfileCheck @Inject() (
       Some(""), Some(""),
       Some(SpotifyRecentlyPlayedInterface.defaultApiEndpoint))
 
+    val userPlaylistsVariant = ApiEndpointVariant(
+      ApiEndpoint("playlists/user", "User's Spotify playlists", None),
+      Some(""), Some(""),
+      Some(SpotifyUserPlaylistsInterface.defaultApiEndpoint))
+
     Seq(
       ApiEndpointVariantChoice("profile", "User's Spotify profile information", active = true, profileVariant),
+      ApiEndpointVariantChoice("playlists/user", "User's Spotify playlists", active = true, userPlaylistsVariant),
       ApiEndpointVariantChoice("feed", "A feed of Spotify tracks played", active = true, recentlyPlayedVariant))
   }
 
+  private def getTracksFromPlaylists(body: Option[JsValue]): Seq[ApiEndpointVariantChoice] = {
+    body.map { responseBody =>
+      (responseBody \ "items").as[Seq[JsValue]] map { playlist =>
```

Comment:
`as` can result in an unhandled error, use of asOpt is preferable (can chain options by using `body.flatMap`.
Instead of casting to Seq[JsValue] can define case class with `id` and `name` and cast to Seq of the case class, more tidy.

Reply:
Can use `SpotifyUsersPlaylist` straight away?